lib/licensee/license.rb

```diff
@@ -157,7 +150,14 @@ def title_regex
         Regexp.new string, 'i'
       end
 
-      parts = [title_regex, key]
+      key_regex = begin
```

Comment:
🤔 are we worried that optimistically matching the key might lead to false positives? In the case of the reference matcher, if I have `# License\n I will sMITe anyone who uses this code`, wouldn't it now match as MIT? Perhaps word boundaries could help?

Reply:
We already have that bug in the reference matcher, see https://github.com/licensee/licensee/issues/392#issuecomment-515782793 ... that's in the current (0.9.11) release, but I suspect has been there for as long as the reference matcher has existed.

I agree that's a bug, but it probably needs to be fixed in the reference matcher.

In the case of MIT, this PR has zero behavior change, since the title and key each end up generating the same regex.